spikeinterface/preprocessing/remove_bad_channels.py

```diff
@@ -1,50 +1,280 @@
+        'x', 'y', 'z', the depth dimension, by default 'y'
+    highpass_filter_cutoff : float
+        If the recording is not filtered, the cutoff frequency of the highpass filter, by default 300
+    n_seconds_for_random_chunks : float
+        Duration of each chunk, by default 0.3
+    n_random_chunks : int
+        Number of random chunks, by default 10
+    seed : int or None
+        The random seed to extract chunks, by default None
 
     Returns
     -------
-    remove_bad_channels_recording: RemoveBadChannelsRecording
-        The recording extractor without bad channels
+    bad_channel_ids : np.array
+        The identified bad channel ids
+    bad_channel_labels : np.array
+        Channels labels depending on the method:
+          * (coeherence+psd) 0: good, 1: dead, 2: noisy, 3: outside of the brain
+          * (std) 0: good, 1 bad
+
+    Examples
+    --------
+
+    >>> import spikeinterface.preprocessing as spre
+    >>> bad_channel_ids, bad_channel_labels = spre.detect_bad_channels(recording, method="coherence+psd")
+    >>> # remove bad channels
+    >>> recording_clean = recording.remove_channels(bad_channel_ids)
+
+    Notes
+    -----
+    For details refer to:
+    International Brain Laboratory et al. (2022). Spike sorting pipeline for the International Brain Laboratory.
+    https://www.internationalbrainlab.com/repro-ephys
     """
-    name = 'remove_bad_channels'
+    method_list = ("std", "coherence+psd")
+    assert method in method_list, f"{method} is not a valid method. Available methods are {method_list}"
+
+    # Get random subset of data to estimate from
+    random_chunk_kwargs = dict(
+        num_chunks_per_segment=n_random_chunks,
+        chunk_size=int(n_seconds_for_random_chunks * recording.sampling_frequency),
+        seed=seed
+    )
+
+    # If recording is not filtered, apply a highpass filter
+    if not recording.is_filtered():
+        recording_hp = highpass_filter(recording, freq_min=highpass_filter_cutoff)
+    else:
+        recording_hp = recording
+
+    # Adjust random chunk kwargs based on method
+    if method == "std":
+        random_chunk_kwargs["return_scaled"] = False
+        random_chunk_kwargs["concatenated"] = True
+    else:
+        random_chunk_kwargs["return_scaled"] = True
+        random_chunk_kwargs["concatenated"] = False
 
-    def __init__(self, recording, bad_threshold=5, **random_chunk_kwargs):
-        random_data = get_random_data_chunks(recording, **random_chunk_kwargs)
+    random_data = get_random_data_chunks(recording_hp, **random_chunk_kwargs)
 
+    if method == "std":
         stds = np.std(random_data, axis=0)
-        thresh = bad_threshold * np.median(stds)
-        keep_inds, = np.nonzero(stds < thresh)
+        thresh = std_threshold * np.median(stds)
+        bad_channel_labels = stds > thresh
+        bad_channel_ids = recording.channel_ids[bad_channel_labels]
+        bad_channel_labels = bad_channel_labels.astype(int)
```

Comment:
Why do we cast bool to int ?

Reply:
I think for consistency with IBL method, that also has numbered labels (they are 0-3 though, 0: good, 1: dead, 2: noisy, 3: outside of the brain). But maybe it is confusing to have bool output in one case and int in other

Reply:
Yep and that is super useful! Because you want to remove label 3 and interpolate the others ;)